tests/api.test.ts

```diff
@@ -277,36 +267,27 @@ describe('request', () => {
     const errorData = {
       error: 'Uh oh, something has gone wrong. Please tweet us @randomapi about the issue. Thank you.',
     };
-    fetch.mockRejectOnce(() => new Promise((_, reject) => setTimeout(() => reject(JSON.stringify(errorData)))), 100);
-    try {
-      await request();
-    } catch (error) {
-      expect(error.message).toBe(errorData.error);
-    }
+    fetch.mockRejectOnce(() => new Promise((_, reject) => setTimeout(() => reject(JSON.stringify(errorData)), 100)));
+    expect(request()).rejects.toThrowError(errorData.error);
   });
 
   it('resolves with function returning object body and init headers', async () => {
-    fetch.mockResponseOnce(() => Promise.resolve({ body: 'ok', init: { headers: { ding: 'dang' } } }), {
-      headers: { bash: 'bang' },
-    });
+    fetch.mockResponseOnce(() => Promise.resolve(new Response('ok', { headers: { ding: 'dang' } })));
```

Comment:
We've lost the second parameter here with `bash: 'bang'`.  It seems like a weird test, and probably points to an api flaw to allow headers to be specified in either the response provider or the params.  (Keep in mind I only forked this, didn't design the api and don't always understand it). 

It looks like from the code you've written, the params will take precedence, whereas previously the test was asserting that the response provider did.  In fact, that seems to be the purpose of `responseInit` in the old code.  I'd like to understand why it was designed this way, and am reluctant to change it for fear that it will break existing tests and cause disruption.

Reply:
Yes, the issue here is that there was a difference between the TypeScript overloads and the actual implementation.

I will revert it so that the original implementation becomes leading, so we won't have breaking changes for non-TypeScript users.